src/NHibernate.Test/NHSpecificTest/NH3957/ResultTransformerEqualityFixture.cs

```diff
@@ -8,55 +7,25 @@ namespace NHibernate.Test.NHSpecificTest.NH3957
 	[TestFixture]
 	public class ResultTransformerEqualityFixture
 	{
-		/// <summary>
-		/// Allows to simulate a hashcode collision. Issue would be unpractical to test otherwise.
-		/// Hashcode collision must be supported for avoiding unexpected and hard to reproduce failures.
-		/// </summary>
-		private void TweakHashcode(System.Type transformerToTweak, object hasher)
-		{
-			var hasherTargetField = transformerToTweak.GetField("Hasher", BindingFlags.Static | BindingFlags.NonPublic);
-			if (!_hasherBackup.ContainsKey(transformerToTweak))
-				_hasherBackup.Add(transformerToTweak, hasherTargetField.GetValue(null));
-
-			// Though hasher is a readonly field, this works at the time of this writing. If it starts breaking and cannot be fixed,
-			// ignore those tests or throw them away.
-			hasherTargetField.SetValue(null, hasher);
-		}
-
-		private Dictionary<System.Type, object> _hasherBackup = new Dictionary<System.Type, object>();
-
-		[SetUp]
-		public void Setup()
-		{
-			var hasherForAll = typeof(AliasToEntityMapResultTransformer)
-				.GetField("Hasher", BindingFlags.Static | BindingFlags.NonPublic)
-				.GetValue(null);
-			TweakHashcode(typeof(DistinctRootEntityResultTransformer), hasherForAll);
-			TweakHashcode(typeof(PassThroughResultTransformer), hasherForAll);
-			TweakHashcode(typeof(RootEntityResultTransformer), hasherForAll);
-			TweakHashcode(typeof(ToListResultTransformer), hasherForAll);
-		}
-
-		[TearDown]
-		public void TearDown()
-		{
-			// Restore those types hashcode. (Avoid impacting perf of other tests, avoid second level query cache
-			// issues if it was holding cached entries (but would mean some tests have not cleaned up properly).)
-			foreach(var backup in _hasherBackup)
-			{
-				TweakHashcode(backup.Key, backup.Value);
-			}
-		}
-
+		public class CustomAliasToEntityMapResultTransformer : AliasToEntityMapResultTransformer { }
```

Comment:
This custom class provides also hashcode collision with current implementaion. Not sure why TweakHashcode was required

Reply:
See #564, it is required to check that two different result transformers will not be considered equal if they happen to have the same hashcode, which was NH-3957 bug. The collision for different instances of the same transformer is intentional and mandatory since they are considered equal.

Removing the tweak causes the `Inequality` suffixed tests to no more test the NH-3957 bug. You should change them to compare with the corresponding `Custom` descendant you have added. That is still a step away from what was NH-3957, but I do not see any other solution with the `Instance` optimization. NH-3957 was a random hashcode collision between different unrelated transformers causing them to wrongly consider themselves equals.

Reply:
>it is required to check that two different result transformers will not be considered equal if they happen to have the same hashcode,.. Removing the tweak causes the Inequality suffixed tests to no more test the NH-3957 bug

But I'm already checking `Inequality`:
[`Assert.False(custom.Equals(transf1));`](https://github.com/nhibernate/nhibernate-core/pull/2000/files#diff-012db71c0b0b1c67464a20c007d76d40R53)
[`Assert.False(transf1.Equals(custom));`](https://github.com/nhibernate/nhibernate-core/pull/2000/files#diff-012db71c0b0b1c67464a20c007d76d40R54)

 I've just put those checks in the wrong tests.
>That is still a step away from what was NH-3957

I still see no difference between inequality check for custom implemented classes with collisions to existing ones ( `custom` and `transf1`) and TweakHashcode hack for existing classes. IMHO they still fully cover the case:
>that two different result transformers will not be considered equal if they happen to have the same hashcode